### PR TITLE
Multiple cameras + compression

### PR DESCRIPTION
**Pull request by brendoncintas**
I noticed that for the demo_launch, the support is mostly for one USB camera. I've modified the launch file to include other cameras, and remapped each `image_raw` and `image_compressed` topic to be on a per-camera basis rather than all of them being on one topic. 

**A note about this:**

- This only works for ROS2. 
- I've commented out the extra usb_cam nodes as I'll be needing 4 for my robot, but you can add or remove as needed.
- For my application, I didn't need the `show_image.py` script, so I removed it from the launch file. This could be added back if needed.

Also, I've cleaned up the code in the launch file to make each camera an individual node within a group, then launch the whole group with LaunchDescription, rather than each node being launched individually. Should make the code look a little cleaner. 

**FROM #247. I didn't PR the right branch. Whoops.**

### CODE REVIEW

**Comment by flynneva** [comment]
@brendoncintas I like this idea of making it easier for users to launch multiple cameras at once. I've taken your branch and made one small improvement to it by making the config into a class that should make it dead simple for people to add new cameras to the launch file.

You can see [the changes I made here](https://github.com/flynneva/usb_cam/commit/a49e4281c239b31c0be4b6cd59bc0639861a77fb).

To add new cameras to the launch file, now all that is needed is to [add a few lines here](https://github.com/flynneva/usb_cam/commit/a49e4281c239b31c0be4b6cd59bc0639861a77fb#diff-f8816fc56a5fa4fe15068a6c668d7f1eb99e5e2deb8ce8c81bdbb605e9d89b1eR57) that adds a new `CameraConfig` class to the list and thats it:

```
CameraConfig(
  name="camera1",
  param_path=Path(path, to, param file),
)
```

If you like them as well, then I can add them to this branch I think and we can go ahead with this PR.

Let me know what you think!

**Comment by brendoncintas** [comment]
> @brendoncintas I like this idea of making it easier for users to launch multiple cameras at once. I've taken your branch and made one small improvement to it by making the config into a class that should make it dead simple for people to add new cameras to the launch file.
> 
> You can see [the changes I made here](https://github.com/flynneva/usb_cam/commit/a49e4281c239b31c0be4b6cd59bc0639861a77fb).
> 
> To add new cameras to the launch file, now all that is needed is to [add a few lines here](https://github.com/flynneva/usb_cam/commit/a49e4281c239b31c0be4b6cd59bc0639861a77fb#diff-f8816fc56a5fa4fe15068a6c668d7f1eb99e5e2deb8ce8c81bdbb605e9d89b1eR57) that adds a new `CameraConfig` class to the list and thats it:
> 
> ```
> CameraConfig(
>   name="camera1",
>   param_path=Path(path, to, param file),
> )
> ```
> 
> If you like them as well, then I can add them to this branch I think and we can go ahead with this PR.
> 
> Let me know what you think!

That is an incredible idea, and I'm all for it. Go ahead and add them to the branch and go ahead with the PR.

File: config/params_2.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    ros__parameters:
+      video_device: "/dev/video2"
+      framerate: 15.0
+      io_method: "mmap"
+      frame_id: "camera"
+      pixel_format: "mjpeg2rgb"  # see usb_cam/supported_formats for list of supported formats
+      image_width: 640
+      image_height: 480
+      camera_name: "test_camera"
+      camera_info_url: "package://usb_cam/config/camera_info.yaml"
```
[the remark this answers is not in the document]
Comment (brendoncintas) [reply config/params_2.yaml:11]: Sounds like a good idea. Also, perhaps a comment in README.md about where to obtain the unique camera intrinsics?

**Review by brendoncintas — COMMENTED** [review]
This should be good to go!

**Comment by flynneva** [comment]
@brendoncintas thanks, I cleaned up the commits and fixed the linter errors. Lets wait for CI to run again as it should pass now

**Review by flynneva — APPROVED** [review]

**Comment by flynneva** [comment]
@brendoncintas thanks for the contribution! 🙏🏼

**Comment by brendoncintas** [comment]
Glad to have been a part of it!

**Pull request merged**